### DISCUSSION TITLE
[v5] How to handle async/await error

### DISCUSSION THREAD

**Discussion by ezehlivinus** [adonisjs/core#2394] [Help]
How can one handle `async/await` error especially but not limited to AdonisJS API app but also its web app?

Currently, this is how I am solving this problem.
I am using `try/catch`, but I don't want to be repeating this everywhere almost in all files.
```
try {
 // some code
const user = await User.create(validatedData)
// some more code
return user;
} catch(error) {
// catch error if any
return error.message
}
```
In any file, I have to repeat this in methods/functions, to make sure I report errors nicely to API consumers.
Sometimes I had to nest `try.catch` like this to handle specific cases.

```
try {
 // some code
const user = await User.find(params.id)
if (!user) {
 return response.status(404).send({ ... })
}
 try {
 // another set of code
 } catch(error) {
 // handle error specific to this try/catch block
}
} catch(error) {
// catch error if any
return error.message
}
```
So all the above are messy
In `expressjs` we solves this problem with an npm package called:[ express-async-errors](https://www.npmjs.com/package/express-async-errors). With it, we write this way:

```
 // some code
const user = await User.create(validatedData)
// some more code without try/catch
return user;
```
Except if the code is away from expressjs request/response (circle), then `try/catch` is used.

So please how can write without repeating `try/catch` everywhere.

**Comment by McSneaky** [comment]
You can create new global error handler and then handle it in there? :thinking:

**Reply by ezehlivinus** [reply]
This error handler you are talking about does it stop me from using try/catch while still writing async/await.

..and do not really need to call the handler in anywhere except once may be in the server/middleware

**Reply by anthrogan** [reply]
Personally I think you should still write try catch, but yes a global error handler will catch all _exceptions_, you could have logic inside of the handler to interoperate the error message and try and response appropriately depending on message. Similar to how the library you shared does.

Strictly speaking though, returning no user isn't really an error. Nothings broken.
